internal/controller/cloudsyncmgr/mqttconfig.go

```diff
@@ -0,0 +1,76 @@
+package cloudsyncmgr
+
+import (
+	"fmt"
+	MQTT "github.com/eclipse/paho.mqtt.golang"
+	"sync"
+	"time"
+)
+
+const clientID = "Homeedge"
+
+// Client is a wrapper on top of `MQTT.Client`
+type Client struct {
+	ID   string
+	Host string
+	Port uint
+	Qos  byte
+	sync.RWMutex
+	ClientOptions *MQTT.ClientOptions
+	MQTT.Client
+}
+
+// Config represents an attribute config setter for the `Client`.
+type Config func(*Client)
+
+// SetClientID sets the mqtt client id.
+func SetClientID(id string) Config {
+	return func(c *Client) {
+		c.ID = id
+	}
+}
+
+// SetHost sets the host where to connect.
+func SetHost(host string) Config {
+	return func(c *Client) {
+		c.Host = host
+	}
+}
+
+// SetPort sets the port where to connect.
+func SetPort(port uint) Config {
+	return func(c *Client) {
+		c.Port = port
+	}
+}
+
+//SetBrokerURL returns the broker url for connection
+func (c *Client) SetBrokerURL(protocol string) string {
+	return fmt.Sprintf("%s://%s:%d", protocol, c.Host, c.Port)
+}
+
+// NewClient returns a configured `Client`. Is mandatory
+func NewClient(configs ...Config) (*Client, error) {
+	client := &Client{
+		Qos: byte(1),
```

Comment:
In future , QOS configuration for MQTT can be supported

Reply:
Currently QoS = 1 is required for our use case scenario. Though its already supported as one can set the byte to 0,2 also using client object